File: ActivityLifeCycle/app/src/main/java/com/vkt4u9999/activitylifecycle/MainActivity.java
package com.vkt4u9999.activitylifecycle;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.os.PersistableBundle;
import android.util.Log;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    TextView textView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        textView=findViewById(R.id.textView);
        textView.append("OnCreate"+ " \n");

        if (savedInstanceState != null){
            textView.setText(savedInstanceState.getString("textToBundle"));
        }
    }



    @Override
    public void onSaveInstanceState(@NonNull Bundle outState) {
        super.onSaveInstanceState(outState);

        textView.append(textView.getText()+"getInstanceState \n");
        outState.putString("textToBundle", textView.getText().toString()+"wqwqww");


    }
}

File: FragmentTrainer/app/src/main/java/com/vkt4u9999/fragmenttrainer/MainActivity.kt
package com.vkt4u9999.fragmenttrainer

import android.annotation.SuppressLint
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import androidx.lifecycle.get
import com.vkt4u9999.fragmenttrainer.databinding.ActivityMainBinding
import com.vkt4u9999.fragmenttrainer.fragments.FragmentA
import com.vkt4u9999.fragmenttrainer.fragments.FragmentB
import com.vkt4u9999.fragmenttrainer.utils.replaceFragment

class MainActivity : AppCompatActivity(){

    private lateinit var mainBinding: ActivityMainBinding
    private lateinit var mainViewModel: MainViewModel



    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        mainBinding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(mainBinding.root)


        replaceFragment(FragmentA())




    }

}

